django/core/management/commands/optimizemigration.py

```diff
@@ -0,0 +1,111 @@
+import sys
+
+from django.apps import apps
+from django.core.management.base import BaseCommand, CommandError
+from django.db import DEFAULT_DB_ALIAS, connections, migrations
+from django.db.migrations.exceptions import AmbiguityError
+from django.db.migrations.loader import MigrationLoader
+from django.db.migrations.optimizer import MigrationOptimizer
+from django.db.migrations.writer import MigrationWriter
+from django.utils.version import get_docs_version
+
+
+class Command(BaseCommand):
+    help = 'Optimizes the operations for the named migration.'
+
+    def add_arguments(self, parser):
+        parser.add_argument('app_label', help='App label of the application containing the migration.')
+        parser.add_argument('migration_name', help='Migration name to optimize the operations for.')
+        parser.add_argument(
+            '--check', action='store_true',
+            help='Exit with a non-zero status if the migration can be optimized.',
+        )
+
+    def handle(self, *args, **options):
+        verbosity = options['verbosity']
+        app_label = options['app_label']
+        migration_name = options['migration_name']
+        check = options['check']
+
+        # Validate app_label
+        try:
+            apps.get_app_config(app_label)
+        except LookupError as err:
+            raise CommandError(str(err))
+
+        # Load the current graph state, check the app and migration they asked for exists
+        loader = MigrationLoader(connections[DEFAULT_DB_ALIAS])
+        if app_label not in loader.migrated_apps:
+            raise CommandError(
+                "App '%s' does not have migrations" % app_label
+            )
+
+        # Find migration
+        try:
+            migration = loader.get_migration_by_prefix(app_label, migration_name)
+        except AmbiguityError:
+            raise CommandError(
+                "More than one migration matches '%s' in app '%s'. Please be "
+                "more specific." % (migration_name, app_label)
+            )
+        except KeyError:
+            raise CommandError(
+                "Cannot find a migration matching '%s' from app '%s'." %
+                (migration_name, app_label)
+            )
+
+        # Optimize migration
+        optimizer = MigrationOptimizer()
+        new_operations = optimizer.optimize(migration.operations, migration.app_label)
+
+        if len(migration.operations) == len(new_operations):
+            if verbosity > 0:
+                self.stdout.write("  No optimizations possible.")
+            return
+        else:
```

Comment:
You added the return statement above if the condition which means no need else statement. You should remove it to make it easier to read.

Reply:
That's not true, `return` is to avoid setting new migrations.

Reply:
Do you think else statement is required in the following example?
```py
def greet():
    condition = False
    if any_condition:
       return "Hi"
    else:
        return "Hello"
```

Reply:
I thought you wanted to remove `return`.  Nevertheless I'd also leave the `else` as it increases readability.

Reply:
I prefer the following one rather than the above one

```
def greet():
    condition = False
    if condition:
       return "Hi"

    return "Hello"
```

Feel free whether follow the things I point out.